### PR TITLE
[MJAVADOC-626] Add a stale javadoc detection mechanism

### PR DESCRIPTION
**Pull request by gnodet**
PR to compute the state of a javadoc generation and skip the generation in case there is no change.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -66,7 +66,7 @@ under the License.
   </distributionManagement>
 
   <properties>
-    <javaVersion>7</javaVersion>
+    <javaVersion>8</javaVersion>
```
Comment (michael-o) [inline pom.xml]: That is not possible because the entire Maven 3 baseline is Java 7.

Comment (gnodet) [reply pom.xml]: 3.x forever on Java 7 ? No plan to at least require java 8 which is 5 years old, while Java 7 has been EOL'ed more than 4 years ago ?

Anyway, if that's really the only objection, I'll work on a different PR, but I'd rather not rewrite the whole code if this ends up being rejected for whatever reason.

Comment (michael-o) [reply pom.xml]: Not for ever, neither an objection, but simply an important point. Java 7 isn't EOL'ed. The tree is alive, but you have to compile your own. The problem simply is that would not work with any previous Maven version where we say Java 7+.

Comment (gnodet) [reply pom.xml]: Ok, i rewrote the usage of streams so that Java 7 only is required.

Comment (olamy) [reply pom.xml]: so ridiculous we are soon in 2020.....

**Comment by gnodet** [comment]
Apart from the java 8 requirement which I removed, any other comment on this PR ?

**Comment by olamy** [comment]
@gnodet all looks good, I will have another last review tomorrow and merge it. Sorry for delay!

**Comment by jira-importer** [comment]
Resolve #919

**Pull request merged**